### PR TITLE
[tests-only] upload or create into folder Shares

### PR DESCRIPTION
**Pull request by ScharfViktor**
A test is written that demonstrates the error. The test will work when the bug is fixed

Description here https://github.com/owncloud/ocis/issues/2322

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/17643/46/1
The following scenarios passed on retry:
- webUILogin/openidLogin.feature:29

**Comment by ownclouders** [comment]
Results for <strong>oC10MoveUpload</strong> https://drone.owncloud.com/owncloud/web/17643/13/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/17645/38/1
The following scenarios passed on retry:
- webUISharingExternal/federationSharing.feature:363

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/17645/64/1
<details><summary>:boom: The acceptance tests failed. Please find the screenshots inside ...</summary>

<p>


### webUIFilesCopy-copy-feature-36.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: tests/acceptance/features/webUICreateFilesFolders/createFolders.feature
Comment (phil-davis) [inline tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:99]:
```suggestion
    Given user "Brian" has been created with default attributes and without skeleton files
    And user "Brian" has created file "lorem.txt"
```

We don't need user Brian in all the scenarios of this feature. If Brian is created in the Background section then it wastes CI time on each of the other scenarios.

File: tests/acceptance/features/webUICreateFilesFolders/createFolders.feature
Comment (phil-davis) [inline tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:8]: This can be removed from here. It is only needed in one scenario, so it the step can go just in that scenario.

**Review by phil-davis — COMMENTED** [review]
Tests are good. This is interesting behavior and good to have tests for the current oC10 behavior. We need everybody to discuss in the issue what the behavior should be.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@ScharfViktor there is now a conflict with the other PR  that was just merged. You need to rebase and sort that out.

**Review by phil-davis — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/17695/27/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/17695/51/1
The following scenarios passed on retry:
- webUIMarkdownEditor/markdownFile.feature:19

**Comment by phil-davis** [comment]
> The test will work when the bug is fixed

The behavior might be a feature, and not need to be fixed.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/17697/33/1
The following scenarios passed on retry:
- webUISharingPublicManagement/publicLinkIndicator.feature:27

File: tests/acceptance/features/webUIUpload/uploadIntoShares.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIUpload/uploadIntoShares.feature:16]:
```suggestion
    When the user opens folder "Shares" 
```

We try to write the `When` "action" step(s) in the present tense.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers1</strong> https://drone.owncloud.com/owncloud/web/17783/54/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:108

File: .drone.star
Comment (phil-davis) [inline .drone.star:624]: There are more places to add this suite name - search for `webUIMoveFilesFolders` - there are groups of pipelines that run with oC10 and with OCIS, and with different typical phone-sized screen resolutions.

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:2]:
```suggestion
  This suite of tests describes different behavior in oCIS and ownCloud10 with respect to 
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:21]:
```suggestion
  Scenario: the Shares folder exists after accepting the first shared file
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:29]:
```suggestion
  Scenario: try to upload a file or a folder into a folder Shares with all permissions in oCIS
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:38]:
```suggestion
  Scenario: try to upload a file or a folder into a folder Shares with read permissions in oCIS
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:79]:
```suggestion
  Scenario: move a file or a folder into a folder Shares in oc10 
```

**Review by phil-davis — COMMENTED** [review]
The tests look good. Just a few suggestions for words in the scenario descriptions.

When you get `.drone.star` sorted out, we will see if the scenarios pass in CI.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPermissions2</strong> https://drone.owncloud.com/owncloud/web/17802/57/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISResharing1</strong> https://drone.owncloud.com/owncloud/web/17802/59/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingBasic</strong> https://drone.owncloud.com/owncloud/web/17803/49/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingBasic</strong> https://drone.owncloud.com/owncloud/web/17823/49/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublicExpireAndRoles</strong> https://drone.owncloud.com/owncloud/web/17839/61/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPermissions2</strong> https://drone.owncloud.com/owncloud/web/17839/57/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/17842/27/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/17842/39/1
The following scenarios passed on retry:
- webUISharingExternalToRoot/federationSharing.feature:356

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/17842/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsers</strong> https://drone.owncloud.com/owncloud/web/17857/24/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md
Comment (phil-davis) [inline tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md:531]: https://github.com/owncloud/ocis/issues/2322 https://github.com/cs3org/reva/pull/1939 should have fixed this?

And I think that code is in oCIS now. But maybe some web code needs to also be adjusted to get these passing.

Anyway, we had better rebase this PR to make sure that we get an up-to-date CI result before merging.

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:21]:
```suggestion
  Scenario: the Shares folder exists after accepting the first shared file
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:29]:
```suggestion
  Scenario: try to upload a file or a folder into a folder Shares with all permissions in oCIS
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:38]:
```suggestion
  Scenario: try to upload a file or a folder into a folder Shares with read permissions in oCIS 
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:68]:
```suggestion
  Scenario: create a new folder into a folder Shares in oc10
```

File: tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature
Comment (phil-davis) [inline tests/acceptance/features/webUIOperationsWithFolderShares/accessToSharesFolder.feature:89]:
```suggestion
  Scenario: try to move a file or a folder into a folder Shares in oCIS
```

**Review by phil-davis — COMMENTED** [review]
This looks good. I will adjust a couple of small English changes, rebase, and it can be merged when CI passes.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/18121/11/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/18121/44/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Basic</strong> https://drone.owncloud.com/owncloud/web/18121/6/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Locks</strong> https://drone.owncloud.com/owncloud/web/18121/7/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/18121/46/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingFolderAdvancedPermissionMU</strong> https://drone.owncloud.com/owncloud/web/18121/18/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublicExpireAndRoles</strong> https://drone.owncloud.com/owncloud/web/18121/61/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/18121/62/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISRename</strong> https://drone.owncloud.com/owncloud/web/18121/48/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/18121/52/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/18121/39/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPermissions1</strong> https://drone.owncloud.com/owncloud/web/18121/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationNotifications</strong> https://drone.owncloud.com/owncloud/web/18121/63/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/18121/35/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone2</strong> https://drone.owncloud.com/owncloud/web/18121/45/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingFilePermission</strong> https://drone.owncloud.com/owncloud/web/18121/17/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18121/27/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/18121/51/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/18121/12/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10CreateDelete</strong> https://drone.owncloud.com/owncloud/web/18121/8/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFavorites</strong> https://drone.owncloud.com/owncloud/web/18121/50/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by phil-davis — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>oC10Basic</strong> https://drone.owncloud.com/owncloud/web/18124/7/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by phil-davis** [comment]
I added `full-ci`  to the PR title, rebased and pushed again so we get a complete set of results. There were some unexpected failures.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18126/24/1
The following scenarios passed on retry:
- webUISharingInternalGroupsToRootSharingIndicator/shareWithGroups.feature:104
- webUISharingInternalGroupsSharingIndicator/shareWithGroups.feature:114

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/18130/39/1
The following scenarios passed on retry:
- webUISharingExternal/federationSharing.feature:363

**Comment by ownclouders** [comment]
Results for <strong>oC10Locks</strong> https://drone.owncloud.com/owncloud/web/18130/8/1
The following scenarios passed on retry:
- webUIWebdavLockProtection/move.feature:123

**Comment by phil-davis** [comment]
I suggest that we rebase this after https://github.com/owncloud/web/pull/5652 is merged - that PR will provide a known-good green CI to the latest oCIS 1.10.0

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait2</strong> https://drone.owncloud.com/owncloud/web/18230/43/1
The following scenarios passed on retry:
- webUIMoveFilesFolders/moveFiles.feature:51

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/18271/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:36

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18309/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersSharingIndicator/shareWithUsers.feature:154

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/18309/52/1
The following scenarios passed on retry:
- webUICreateFilesFolders/createFolderEdgeCases.feature:19

**Comment by phil-davis** [comment]
The linked drone build https://drone.owncloud.com/owncloud/web/18313/1/1 shows that it was cancelled after 5 seconds.
I have no idea why that happened.
I will restart it.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/18321/45/1
The following scenarios passed on retry:
- webUIDeleteFilesFolders/deleteFilesFolders.feature:59

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/18321/40/1
The following scenarios passed on retry:
- webUISharingExternalToRoot/federationSharing.feature:356

**Comment by phil-davis** [comment]
The usual 0C10Basic pipeline that passes in CI has:
```
$ cucumber-js --retry 1 --require-module @babel/register --require-module @babel/polyfill --require tests/acceptance/setup.js --require tests/acceptance/stepDefinitions --format node_modules/cucumber-pretty tests/acceptance/features/webUIAccount tests/acceptance/features/webUILogin tests/acceptance/features/webUIPreview tests/acceptance/features/webUIPrivateLinks -t 'not @skip and not @skipOnOC10 and not @openIdLogin'
```

That has `not @openIdLogin` in the filter tags. Those tests are skipped when running against oC10.

But https://drone.owncloud.com/owncloud/web/18321/7/14 has only `not @skipOnOC10 and not @notToImplementOnOC10` in the filter tags. So those OpenId test scenarios were being run.

I added a commit to fix that.

File: nodesource_setup.sh
Comment (phil-davis) [inline nodesource_setup.sh:3]: Why is this file committed?
I will delete it...

**Review by phil-davis — CHANGES_REQUESTED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18335/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/18335/12/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Pull request merged**
